### PR TITLE
fix: enable errcheck linter

### PR DESCRIPTION
**Pull request by garethjevans**

### CODE REVIEW

File: src/autoscaler/db/sqldb/appmetric_sqldb.go [hunk 1]
```diff
@@ -83,35 +83,38 @@ func (adb *AppMetricSQLDB) SaveAppMetricsInBulk(appMetrics []*models.AppMetric)
...
+			_ = txn.Rollback()
 			return err
 		}
 
 	case "mysql":
 		sqlStr := "INSERT INTO app_metric(app_id,metric_type,unit,timestamp,value)VALUES"
 		vals := []interface{}{}
 		if len(appMetrics) == 0 {
-			txn.Rollback()
+			err = txn.Rollback()
```
Comment (KevinJCross) [inline src/autoscaler/db/sqldb/appmetric_sqldb.go:114]: not sure what this is about ?
I think you can just put _ = ?

File: src/autoscaler/eventgenerator/aggregator/aggregator.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func (a *Aggregator) startSavingAppMetric() {
 		case <-ticker.C():
 			if len(appMetricArray) > 0 {
 				go func(appMetricDB db.AppMetricDB, metrics []*models.AppMetric) {
-					appMetricDB.SaveAppMetricsInBulk(metrics)
+					_ = appMetricDB.SaveAppMetricsInBulk(metrics)
```
Comment (KevinJCross) [inline src/autoscaler/eventgenerator/aggregator/aggregator.go:109]: maybe we should log this error ?

File: src/autoscaler/metricsgateway/helpers/ws_helper_test.go [hunk 3]
```diff
@@ -108,10 +108,9 @@ var _ = Describe("WsHelper", func() {
 			err = wsHelper.Ping()
 			Expect(err).ShouldNot(HaveOccurred())
 			Eventually(pingPongChan, 5*time.Second, 1*time.Second).Should(Receive(Equal(1)))
-
 		})
 		It("close the websocket connection", func() {
-			wsHelper.CloseConn()
+			err = wsHelper.CloseConn()
 			Expect(err).NotTo(HaveOccurred())
```
Comment (KevinJCross) [inline src/autoscaler/metricsgateway/helpers/ws_helper_test.go:114]: this was checking the error that was set in errr before the action ... maybe we should do 2 checks ?

File: src/autoscaler/db/sqldb/appmetric_sqldb.go (hunk 1 above)
Comment (garethjevans) [reply src/autoscaler/db/sqldb/appmetric_sqldb.go:114]: I've updated it to `return err`

File: src/autoscaler/metricsserver/collector/collector.go [hunk 4]
```diff
@@ -214,7 +214,7 @@ func (c *metricCollector) saveMetrics() {
 		case <-ticker.C():
 			if c.PersistMetrics && len(metrics) > 0 {
 				go func(instancemetricsDb db.InstanceMetricsDB, metrics []*models.AppInstanceMetric) {
-					instancemetricsDb.SaveMetricsInBulk(metrics)
+					_ = instancemetricsDb.SaveMetricsInBulk(metrics)
```
Comment (KevinJCross) [inline src/autoscaler/metricsserver/collector/collector.go:217]: same as the other comment

File: src/autoscaler/scalingengine/scalingengine.go [hunk 5]
```diff
@@ -68,7 +68,9 @@ func (s *scalingEngine) Scale(appId string, trigger *models.Trigger) (*models.Ap
 		Reason:       getDynamicScalingReason(trigger),
 	}
 
-	defer s.scalingEngineDB.SaveScalingHistory(history)
+	defer func() {
+		_ = s.scalingEngineDB.SaveScalingHistory(history)
```
Comment (KevinJCross) [inline src/autoscaler/scalingengine/scalingengine.go:72]: hmmm should we log the error here ?

Comment (KevinJCross) [reply src/autoscaler/scalingengine/scalingengine.go:72]: or maybe SaveScalingHistory should log it if it does not already

File: src/autoscaler/metricsgateway/helpers/ws_helper_test.go (hunk 3 above)
Comment (garethjevans) [reply src/autoscaler/metricsgateway/helpers/ws_helper_test.go:114]: the previous error is checked in the `BeforeEach` func

File: src/autoscaler/scalingengine/scalingengine.go (hunk 5 above)
Comment (KevinJCross) [reply src/autoscaler/scalingengine/scalingengine.go:72]: it logs it anyway

Comment (garethjevans) [reply src/autoscaler/scalingengine/scalingengine.go:72]: done

**Review by KevinJCross — APPROVED** [review]

**Pull request merged**